### ISSUE TITLE
Implement support for nested query arguments

### ISSUE THREAD

**Issue by Kashuab** [Kashuab/mobx-depot#34]
We need to implement the ability to, within a selection builder, provide arguments for nested queries:

```graphql
{
  user(id: 1234) {
    email
    name  
    posts(first: 10, sort: "ascending") {
      title
      content
    }
  }
}
```

I'm not sure what the most intuitive approach would be. Here's the first thought that comes to mind:

```ts
const userQuery = new UserQuery(
  { id: 1234 },
  user => user
    .email
    .name
    .posts.first(10).sort("ascending")(
      post => post.title.content
    )
);
```

**Comment by Kashuab** [comment]
Implemented, published version 0.0.42

**Comment by Kashuab** [comment]
Actually, changed my mind. Instead of chaining functions, you can now pass an `args` argument to the query. This allows for support of required args:

```ts
const userQuery = new UserQuery(
  { id: 1234 },
  user => user
    .email
    .name
    .posts(
      { first: 10, sort: "ascending" },
      post => post.title.content
    )
);
```
